Title: Python Documentation Index
URL: https://docs.python.org/3/genindex-all.html
Summary: This section lists various attributes and functions related to the doctest and traceback modules in Python, with hyperlinks to the corresponding documentation sections.
---

doctest.UnexpectedException attribute)](https://docs.python.org/3/library/doctest.html#doctest.UnexpectedException.example)
  * [examples (doctest.DocTest attribute)](https://docs.python.org/3/library/doctest.html#doctest.DocTest.examples)
  * [exc_info (doctest.UnexpectedException attribute)](https://docs.python.org/3/library/doctest.html#doctest.UnexpectedException.exc_info)
    * [(in module sys)](https://docs.python.org/3/c-api/intro.html#index-13), [[1]](https://docs.python.org/3/reference/datamodel.html#index-68)
  * [exc_info() (in module sys)](https://docs.python.org/3/library/sys.html#sys.exc_info)
  * [exc_msg (doctest.Example attribute)](https://docs.python.org/3/library/doctest.html#doctest.Example.exc_msg)
  * [exc_type (traceback.TracebackException attribute)](https://docs.python.org/3/library/traceback.html#traceback.TracebackException.exc_type)
  * [exc_type_str (traceback.TracebackException attribute)](https://docs.python.org/3/library/traceback.html#traceback.TracebackException.exc_type_str)
  * [excel (class in csv)](https://docs.python.org/3/library/csv.html#csv.excel)
  * [excel_tab (class in csv)](https://docs.python.org/3/library/csv.html#csv.excel_tab)
  * except 
    * [keyword](https://docs.python.org/3/reference/compound_stmts.html#index-9)
    * [statement](https://docs.python.org/3/library/exceptions.html#index-0)
  * except_star 
    * [keyword](https://docs.python.org/3/reference/compound_stmts.html#index-12)
  * [ExceptHandler (class in ast)](https://docs.python.org/3/library/ast.html#ast.ExceptHandler)
  * [excepthook() (in module sys)](https://docs.python.org/3/library/sys.html#sys.excepthook)
    * [(in module threading)](https://docs.python.org/3/library/threading.html#threading.excepthook)
  * [Exception](https://docs.python.org/3/library/exceptions.html#Exception)
  * [exception](https://docs.python.org/3/reference/executionmodel.html#index-15), [[1]](https://docs.python.org/3/reference/simple_stmts.html#index-27)
    * [AssertionError](https://docs.python.org/3/reference/simple_stmts.html#index-19)
    * [AttributeError](https://docs.python.org/3/reference/expressions.html#index-40)
    * [chaining](https://docs.python.org/3/library/exceptions.html#index-2), [[1]](https://docs.python.org/3/reference/simple_stmts.html#index-29)
    * [GeneratorExit](https://docs.python.org/3/reference/expressions.html#index-33), [[1]](https://docs.python.org/3/reference/expressions.html#index-37)
    * [handler](https://docs.python.org/3/reference/datamodel.html#index-68)
    * [ImportError](https://docs.python.org/3/reference/simple_stmts.html#index-34)
    * [NameError](https://docs.python.org/3/reference/expressions.html#index-4)
    * [raising](https://docs.python.org/3/reference/simple_stmts.html#index-27)
    * [StopAsyncIteration](https://docs.python.org/3/reference/expressions.html#index-36)
    * [StopIteration](https://docs.python.org/3/reference/expressions.html#index-32), [[1]](https://docs.python.org/3/reference/simple_stmts.html#index-26)
    * [TypeError](https://docs.python.org/3/reference/expressions.html#index-64)
    * [ValueError](https://docs.python.org/3/reference/expressions.html#index-73)
    * [ZeroDivisionError](https://docs.python.org/3/reference/expressions.html#index-68)
  * [EXCEPTION (in module _tkinter)](https://docs.python.org/3/library/tkinter.html#tkinter.EXCEPTION)
  * [exception handler](https://docs.python.org/3/reference/executionmodel.html#index-16)
  * [exception() (asyncio.Future method)](https://docs.python.org/3/library/asyncio-future.html#asyncio.Future.exception)
    * [(asyncio.Task method)](https://docs.python.org/3/library/asyncio-task.html#asyncio.Task.exception)
    * [(concurrent.futures.Future method)](https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.Future.exception)
    * [(in module logging)](https://docs.python.org/3/library/logging.html#logging.exception)
    * [(in module sys)](https://docs.python.org/3/library/sys.html#sys.exception)
    * [(logging.Logger method)](https://docs.python.org/3/library/logging.html#logging.Logger.exception)
  * [EXCEPTION_HANDLED (monitoring event)](https://docs.python.org/3/library/sys.monitoring.html#monitoring-event-EXCEPTION_HANDLED)
  * [ExceptionGroup](https://docs.python.org/3/library/exceptions.html#ExceptionGroup)
  * [exceptions (BaseExceptionGroup attribute)](https://docs.python.org/3/library/exceptions.html#BaseExceptionGroup.exceptions)
    * [(pdb command)](https://docs.python.org/3/library/pdb.html#pdbcommand-exceptions)
    * [(traceback.TracebackException attribute)](https://docs.python.org/3/library/traceback.html#traceback.TracebackException.exceptions)
  * [EXCLAMATION (in module token)](https://docs.python.org/3/library/token.html#token.EXCLAMATION)
  * exclusive 
    * [or](https://docs.python.org/3/reference/expressions.html#index-76)
  * [EXDEV (in module errno)](https://docs.python.org/3/library/errno.html#errno.EXDEV)
  * exec 
    * [built-in f